chrome_extension_starfish_ide_from_scratch/content.js

```diff
@@ -0,0 +1,57 @@
+
+current_url=window.location.href;
+var page_uri ="ide://#"+window.location.href;
+ 
+if(current_url.indexOf("https://github.com/")!=-1)
+
+{//Code for Github Begins
+
+  if(!localStorage.getItem("starfish")){
+     navigator.registerProtocolHandler("web+ide",
+                           "https://github.com/?ide=%s",
```

Comment:
not following what this register does - as I understand it this register "ide://<url>" to just redirect to "https://github.com/?ide=<url>" ?

Reply:
The Navigator method registerProtocolHandler() lets web sites register their ability to open or handle particular URL schemes (aka protocols).

For example, this API lets webmail sites open mailto: URLs, or VoIP sites open tel: URLs.

It registers protocol handlers in the browser.
You could see all of the registered handlers here: **[ chrome://settings/handlers](chrome://settings/handlers)**